stixcore/products/level0/quicklookL0.py

```diff
@@ -150,13 +153,19 @@ def from_levelb(cls, levelb, parent=''):
         control.add_data('compression_scheme_triggers_skm',
                          _get_compression_scheme(packets, 'NIX00274'))
 
-        triggers = packets.get_value('NIX00274').T
-        triggers_var = packets.get_value('NIX00274', attr="error").T
-        if control['compression_scheme_triggers_skm'].tolist() == [[0, 0, 7]]:
-            logger.debug('Unscaling trigger ')
-            triggers, triggers_var = unscale_triggers(
-                triggers, integration=duration,
-                detector_masks=control['detector_mask'], ssid=levelb.ssid)
+        triggers = packets.get_value('NIX00274').flatten()
+        triggers_var = packets.get_value('NIX00274', attr="error").flatten()
+
+        cur_index = 0
+        for i, (num, skm) in enumerate(zip(control['num_samples'],
+                                           control['compression_scheme_triggers_skm'])):
+            if skm.tolist() == _SKM_SCALING:
+                cur_slice = slice(cur_index, cur_index+num)
+                logger.debug('Unscaling triggers')
+                triggers[cur_slice], triggers_var[cur_slice] = unscale_triggers(
+                    triggers[cur_slice], integration=duration[cur_slice],
+                    detector_masks=control['detector_mask'][i:i+1], ssid=levelb.ssid)
+            cur_index += num
```

Comment:
Possible optimisation and could be extended for the other science data is rather than looping over all packets loop over data where skm apply usually only 1 skm and in the vast majority of cases number of different skm << num of packets.